src/main/scala/com/databricks/labs/overwatch/env/Workspace.scala

```diff
@@ -169,19 +180,27 @@ class Workspace(config: Config) extends SparkSessionWrapper {
       "end_value" -> s"${untilTimeMs}",
       "increment_counter" -> "3600000",
       "final_response_count" -> s"${finalResponseCount}",
-      "result_key" -> "res"
+      "result_key" -> "res",
+      "tmp_success_path" -> tmpClusterEventsSuccessPath,
+      "tmp_error_path" -> tmpClusterEventsErrorPath
     )
 
     // calling function to make parallel API calls
     val apiCallV2Obj = new ApiCallV2(config.apiEnv)
-    val tmpSqlQueryHistorySuccessPath= apiCallV2Obj.makeParallelApiCalls(sqlQueryHistoryEndpoint, jsonInput, config)
+    val tmpSqlQueryHistorySuccessPath= apiCallV2Obj.makeParallelApiCalls(sqlQueryHistoryEndpoint, jsonInput, pipelineSnapTime.asUnixTimeMilli,config)
     logger.log(Level.INFO, " sql query history landing completed")
 
-    if(Helpers.pathExists(tmpSqlQueryHistorySuccessPath)) {
+    if (Helpers.pathExists(tmpSqlQueryHistorySuccessPath)) {
       try {
-        spark.read.json(tmpSqlQueryHistorySuccessPath)
-          .select(explode(col("res")).alias("res")).select(col("res" + ".*"))
-          .withColumn("organization_id", lit(config.organizationId))
+        val rawDF = deriveRawApiResponseDF(spark.read.json(tmpSqlQueryHistorySuccessPath))
+        if (rawDF.columns.contains("res")) {
+          rawDF.select(explode(col("res")).alias("res")).select(col("res" + ".*"))
+            .withColumn("organization_id", lit(config.organizationId))
+        } else {
+          println(s"""No Data is present for sql/query/history from - ${fromTimeMs} to - ${untilTimeMs},res column not found in dataset""")
```

Comment:
purpose of println is to show the no Data issue directly in front of the console. It will serve the purpose to quickly know if we see no data for sql/query/history. This code is already present in 722 release.